lib/Conch/Command/insert_validation_states.pm

```diff
@@ -0,0 +1,153 @@
+package Conch::Command::insert_validation_states;
+
+=pod
+
+=head1 NAME
+
+insert_validation_states - insert new validation_state records from old device_validate data
+
+=head1 SYNOPSIS
+
+    bin/conch insert_validation_states [long options...] inputfile.csv
+
+        --help  print usage message and exit
+
+=cut
+
+use Mojo::Base 'Mojolicious::Command', -signatures;
+use Getopt::Long::Descriptive;
+use Text::CSV_XS;
+
+has description => 'Insert new validation_state records from old device_validate data';
+
+has usage => sub { shift->extract_usage };  # extracts from SYNOPSIS
+
+sub run ($self, @opts) {
+
+    local @ARGV = @opts;
+    my ($opt, $usage) = describe_options(
+        # the descriptions aren't actually used anymore (mojo uses the synopsis instead)... but
+        # the 'usage' text block can be accessed with $usage->text
+        'insert_validation_states %o inputfile.csv',
+        [],
+        [ 'help', 'print usage message and exit', { shortcircuit => 1 } ],
+    );
+
+    my $start_time = time;
+    my $not_found = 0;
+    my $infile = shift @ARGV or die $opt->usage;
+
+    # do work inside a transaction, in case there is a problem...
+    my $schema = $self->app->schema;
+
+    my $rows_created = $schema->txn_do(sub {
+        my $csv = Text::CSV_XS->new({ binary => 1, eol => $/ });
+        open my $fh, $infile or die "cannot open $infile for reading: $!";
+
+        # document has no header line, and lines look like:
+        # uuid,some-serial-number,1,2018-03-31 19:58:49.604249-04
+        # i.e. report_id, device_id, device_validate.validation->'status', created
+
+        my $device_report_rs = $schema->resultset('device_report');
+        my $validation_state_rs = $schema->resultset('validation_state');
+
+        my $rows_created = 0;
+        my %exists_report_ids;
+        my %not_exists_report_ids;
+
+        my $validation_plan = $schema->resultset('validation_plan')->update_or_create({
+            name => 'Conch v1 placeholder',
```

Comment:
I suggest making the name here something a bit more clear like 'device_validate placeholder' since we cut v2 when the mojo-ization happened, not when we changed validation systems.

Not a show stopper though.

Reply:
ok, will do that now